module/VuFind/src/VuFind/ChannelProvider/SimilarItems.php

```diff
@@ -117,8 +117,16 @@ public function __construct(
      */
     public function setOptions(array $options)
     {
-        $this->batchSize = $options['batchSize'] ?? 24;
         $this->maxRecordsToExamine = $options['maxRecordsToExamine'] ?? 2;
+
+        // Calculate batch size
+        $itemsPerRow = $options['itemsPerRow'] ?? 6;
+        $rowsPerPage = $options['rowsPerPage'] ?? 2;
+        $this->batchSize = $itemsPerRow * $rowsPerPage;
+        // Set a minimum of 20 to make sure the server isn't hit too often
+        while ($this->batchSize <= 20) {
+            $this->batchSize *= 2;
+        }
```

Comment:
Since this logic is repeated so many times, I wonder if we should create a VuFind\ChannelProvider\Feature\BatchTrait that defines the batchSize property and contains a setBatchSizeFromOptions() method, and then refactor all of the impacted code to use it. Other batch-related logic could go there too if appropriate.